src/libs/ajax/User.test.ts

```diff
@@ -1,38 +1,441 @@
+}));
+
+const completeUserProfile: TerraUserProfile = {
+  firstName: 'testFirstName',
+  lastName: 'testFirstName',
+  contactEmail: 'testFirstName',
+  department: 'testDepartment',
+  institute: 'testInstitute',
+  interestInTerra: 'testInterestInTerra',
+  programLocationCity: 'testCity',
+  programLocationCountry: 'testCountry',
+  programLocationState: 'testState',
+  starredWorkspaces: 'testStarredWorkspaces',
+  title: 'testTitle',
+  researchArea: 'testResearchArea',
+};
+
+const minimalCreateUserProfileRequest: CreateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+};
+
+const completeCreateUserProfileRequest: CreateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+  department: completeUserProfile.department,
+  institute: completeUserProfile.institute,
+  interestInTerra: completeUserProfile.interestInTerra,
+  title: completeUserProfile.title,
+};
+
+const minimalUpdateUserProfileRequest: UpdateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+};
+
+const completeUpdateUserProfileRequest: UpdateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+  department: completeUserProfile.department,
+  institute: completeUserProfile.institute,
+  interestInTerra: completeUserProfile.interestInTerra,
+  title: completeUserProfile.title,
+  programLocationCity: completeUserProfile.programLocationCity,
+  programLocationState: completeUserProfile.programLocationState,
+  programLocationCountry: completeUserProfile.programLocationCountry,
+  researchArea: completeUserProfile.researchArea,
+};
+
+const NA = 'N/A';
+
+// create user profile
+describe('A create user profile request', () => {
+  describe('when minimally filled out by the user', () => {
+    // when a user has filled out the required fields of a profile
+    // and has not entered values for the optional fields
+    // those optional fields should be changed to 'N/A' in the request
+    it.each(['title', 'institute', 'programLocationCity', 'programLocationState', 'programLocationCountry'])(
+      'sets undefined property %s to "N/A"',
+      async (field) => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody[field]).toBe(NA);
+      }
+    );
+    describe('should not modify property', () => {
+      // these required fields that the user filled out should not be changed
+      it('interestInTerra', async () => {
```

Comment:
Nit: looks like there's several more opportunities in this file to remove duplicate code using `it.each`.

```suggestion
    it.each(['interestInTerra', ...')('should not modify property %s', (field) => {
```

Reply:
I had originally chosen not to collapse some of these assertions using it.each because although they're the same in our mock here, they can't be assumed to be the same in all cases, but I went back and forth about it so for brevity I can condense more

Reply:
actually looking at it again, are you thinking that all the cases with the same check (e.g. all fields that should be `N/A` or `undefined`) would be collapsed with it.each? Because the logic is quite opaque as to which fields are required, which are required to be non-null, and which are expected to be undefined, I would prefer not to combine those under it.each since they are technically under different requirements